### PR TITLE
VASP: Add uenv tests for GH200

### PR DESCRIPTION
**Pull request by AdhocMan**
Compilation and testing of VASP on GH200 using uenv.
The matching config file for the uenv is added in https://github.com/eth-cscs/alps-uenv/pull/175.

Similar to #245, `CSCS_REGISTRY_USERNAME` and `CSCS_REGISTRY_PASSWORD` need to be set to pull the uenv image.

### CODE REVIEW

File: checks/apps/vasp/vasp_check.py [hunk 1]
```diff
@@ -1,124 +1,179 @@
...
+    def prepare_run(self):
+        self.uarch = uenv.uarch(self.current_partition)
+        config = slurm_config[self.test_name][self.uarch]
+        # sbatch options
+        self.job.options = [
+            f'--nodes={self.num_nodes}',
+        ]
+        self.num_tasks_per_node = config['ntasks-per-node']
+        self.num_tasks = self.num_nodes * self.num_tasks_per_node
+        self.num_cpus_per_task  = config['cpus-per-task']
```
Comment (jgphpc) [inline checks/apps/vasp/vasp_check.py:56]:
```suggestion
        self.num_cpus_per_task = config['cpus-per-task']
```

File: checks/apps/vasp/vasp_check.py [hunk 2]
```diff
@@ -1,124 +1,179 @@
...
+        if self.uarch is not None and \
+           self.uarch in vasp_references[self.test_name]:
+            self.reference = {
+                self.current_partition.fullname:
+                    vasp_references[self.test_name][self.uarch][self.num_nodes]
             }
-        }
-    }
 
+
```
Comment (jgphpc) [inline checks/apps/vasp/vasp_check.py:84]:
```suggestion
```

File: checks/apps/vasp/vasp_check.py [hunk 3]
```diff
@@ -1,124 +1,179 @@
...
+        self.build_time_limit = '0d1h0m0s'
+
+        makefile = f'makefile.include.{self.uarch}'
+        makefile_path = os.path.join(self.prefix, self.sourcesdir, makefile)
+
+        if not os.path.isfile(makefile_path):
+            self.skip(f'No makefile for uarch {self.uarch}')
+
+
+        vasp_download_cmd = f'curl --retry 5 -u ${{CSCS_REGISTRY_USERNAME}}:${{CSCS_REGISTRY_PASSWORD}} -X GET https://jfrog.svc.cscs.ch/artifactory/uenv-sources/vasp/vasp-{self.version}.tar.bz2 -o vasp_src.tar.bz2'
```
Comment (jgphpc) [inline checks/apps/vasp/vasp_check.py:138]: 1/ we need to give the user reframe (gid=djenkssl) permission to read:
https://jfrog.svc.cscs.ch/ui/repos/tree/EffectivePermission/uenv-sources/vasp/vasp-6.4.3.tar.bz2

2/ vasp_download_cmd is too long, break it down in multiplines please.

File: checks/apps/vasp/vasp_check.py [hunk 4]
```diff
@@ -1,124 +1,179 @@
...
+
+        # Download source and copy makfile matching uarch
+        self.prebuild_cmds = [
+            vasp_download_cmd,
+            'tar -xf vasp_src.tar.bz2',
+            # The vasp tar ball contains inconsistent directory names between versions,
```
Comment (jgphpc) [inline checks/apps/vasp/vasp_check.py:144]:
```suggestion
            # The vasp tar ball contains inconsistent directory names between
            # versions ...
```

File: checks/apps/vasp/vasp_check.py [hunk 5]
```diff
@@ -1,124 +1,179 @@
...
+            'find . -maxdepth 1 -type d -name "vasp*" -exec mv {} vasp_src \;',
+            'cd vasp_src',
+            f'cp ../{makefile} makefile.include'
+        ]
+
+        self.vasp_std_executable = os.path.join(
+            self.stagedir, 'vasp_src', 'bin', 'vasp_std'
+        )
+
+
```
Comment (jgphpc) [inline checks/apps/vasp/vasp_check.py:155]:
```suggestion
```

File: checks/apps/vasp/vasp_check.py [hunk 6]
```diff
@@ -1,124 +1,179 @@
...
-        self.job.launcher.options = ['--cpu-bind=cores']
-        self.job.options = ['--distribution=block:block']
-        self.num_tasks_per_node = proc.num_sockets
-        self.num_cpus_per_task = proc.num_cores // self.num_tasks_per_node
-        self.num_tasks = self.num_nodes * self.num_tasks_per_node
-        self.env_vars = {
-            'OMP_NUM_THREADS': self.num_cpus_per_task,
-            'OMP_PLACES': 'cores',
-            'OMP_PROC_BIND': 'close'
-        }
```
Comment (jgphpc) [inline checks/apps/vasp/vasp_check.py:170]:
```suggestion
```

File: checks/apps/vasp/vasp_check.py [hunk 7]
```diff
@@ -1,124 +1,179 @@
...
+
+        vasp_download_cmd = f'curl --retry 5 -u ${{CSCS_REGISTRY_USERNAME}}:${{CSCS_REGISTRY_PASSWORD}} -X GET https://jfrog.svc.cscs.ch/artifactory/uenv-sources/vasp/vasp-{self.version}.tar.bz2 -o vasp_src.tar.bz2'
+
+        # Download source and copy makfile matching uarch
+        self.prebuild_cmds = [
+            vasp_download_cmd,
+            'tar -xf vasp_src.tar.bz2',
+            # The vasp tar ball contains inconsistent directory names between versions,
+            # so we find the directory name and change it to vasp_src
+            'find . -maxdepth 1 -type d -name "vasp*" -exec mv {} vasp_src \;',
```
Comment (jgphpc) [inline checks/apps/vasp/vasp_check.py:146]:
```suggestion
            'find . -maxdepth 1 -type d -name "vasp*" -exec mv {} vasp_src \\;',
```
I get `W605 invalid escape sequence '\;'` with pycodestyle.

File: checks/apps/vasp/src/makefile.include.gh200 [hunk 8]
```diff
@@ -0,0 +1,122 @@
+# Default precompiler options
```
Comment (jgphpc) [inline checks/apps/vasp/src/makefile.include.gh200:1]: would it be possible to add a README with some info about the input files (or is that private info) ?

Comment (AdhocMan) [reply checks/apps/vasp/src/makefile.include.gh200:1]: VASP comes with a set of makefiles for different architectures. This file is just taken from there and slightly modified to include the correct link paths. The other input files are unchanged and originate from https://www.nsc.liu.se/~pla/vasptest, as commented in the `INCAR` file. Not sure if that's enough to put into a readme.

**Pull request merged**